services/rpcfilters/api.go

```diff
@@ -135,15 +183,23 @@ func (api *PublicAPI) UninstallFilter(id rpc.ID) bool {
 // last time it was called. This can be used for polling.
 //
 // https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges
-func (api *PublicAPI) GetFilterChanges(id rpc.ID) ([]common.Hash, error) {
+func (api *PublicAPI) GetFilterChanges(id rpc.ID) (interface{}, error) {
 	api.filtersMu.Lock()
 	defer api.filtersMu.Unlock()
 
 	if f, found := api.filters[id]; found {
-		return f.PopHashes(), nil
+		deadline := f.deadline()
+		if deadline != nil {
+			if !deadline.Stop() {
+				// timer expired but filter is not yet removed in timeout loop
+				// receive timer value and reset timer
+				<-deadline.C
+			}
+			deadline.Reset(api.filterLivenessPeriod)
+		}
+		return f.pop(), nil
 	}
-
-	return []common.Hash{}, errors.New("filter not found")
+	return []interface{}{}, errors.New("filter not found")
```

Comment:
`nil` can be returned instead. I think we discussed it to once that we return an empty slice to stay compatible...?

Reply:
yes, tbh i don't know why it is done this way. but this api and geth are intentionally creating empty slices if there is nothing to return